Rulebook 3.0 (#113)
International-Data-Spaces-Association/IDSA-Rulebook: 010_Planes.md

### Examples of Data Planes

- **RESTful APIs**: A common data plane implementation using HTTP-based protocols for synchronous data retrieval. 

- **Message Queues (e.g., AMQP or MQTT)**: Asynchronous data transmission via publish-subscribe models. 

- **Peer-to-Peer Protocols (e.g., IPFS or BitTorrent-like systems)**: Decentralized file sharing without central servers.

- **Streaming Protocols (e.g., Kafka or WebSockets)**: Continuous data streams with subscriptions. It handles high-volume, low-latency flows while maintaining separation from control plane governance.

These examples assume minimal shared semantics for interoperability. Implementations must account for failure modes, such as network failures or policy violations.

Sovereign capabilities depend on the specific implementation of a data plane and the underlying data transfer technology.

## Data Management Plane

Once data has been shared it usually needs to be managed to enforce governance models and ensure the adherance to policies negotiated in the control plane.

> **The Data Management Plane is not a part of the data space**, but rather a necessary organisational capability, it might have to be aware of data space elements to enable the organisation to fulfil obligations associated with the shared data or to enable the management of data to be shared with others.

Reviewer: I would tone this down a notch.

```suggestion
The Data Management Plane encompasses the organisational functions responsible for managing the lifecycle, quality, and governance of data within a data sharing solution. This includes ensuring data integrity, compliance with contractual obligations, and usability across internal systems.
```

The Data Management Plane encompasses the functions responsible for the lifecycle, quality, and governance of data within a data sharing solution.

> It operates independently of data transmission (data plane) and orchestration (control plane), focusing on ensuring data integrity, compliance, and usability. In data spaces, it supports integration with governance mechanisms across the data user organisation.

Reviewer: ```suggestion
While it operates separately from data transmission (data plane) and orchestration (control plane), the Data Management Plane supports participation in the data space by integrating governance requirements into the organisation’s internal processes and systems.
```

### Typical Functions of Data Management

- **Data Ingestion and Storage**: Collecting, validating, and storing data from various sources, ensuring format consistency and metadata attachment. It enforces invariants like data provenance and traceability to support sovereignty.

- **Data Processing and Transformation**: Applying algorithms for cleansing, aggregation, or anonymization, while adhering to policies reconciled in the control plane. 

- **Data Cataloging and Discovery**: Maintaining metadata registries for asset discovery, ensuring that policies from data sharing contracts are being enforced. 

### Typical Functions of Data Governance

- **Policy Enforcement and Compliance**: Monitoring adherence to governance rules, such as retention policies or usage constraints, through runtime checks. 

- **Quality Assurance and Auditing**: Assessing data accuracy, completeness, and lineage, with audit trails for trust verification. 

- **Risk Management and Data Sovereignty Protection**: Identifying and mitigating risks, including sovereignty violations, by enforcing participant controls. It prioritises conservative approaches, such as deny-by-default for sensitive data.

> These functions are clearly separated from other planes. However, interaction and integration with other planes is necessary for building full governance and mangement capabilities.

Reviewer: ```suggestion
These functions are clearly separated from other planes. However, interaction and integration with other planes is necessary for building full governance and management capabilities.
```
